python-avd/pyavd/api/ip_addressing/utils.py

```diff
@@ -200,7 +200,7 @@ def _get_downlink_ipv6_pool_and_offset(self: AvdIpAddressingProtocol, uplink_swi
         )
         raise AristaAvdError(msg)
 
-    def _get_p2p_ipv4_pool_and_offset(self: AvdIpAddressingProtocol, uplink_switch_index: int) -> tuple[str, int]:
+    def _get_p2p_ipv4_pool_and_offset(self: AvdIpAddressingProtocol, uplink_switch_index: int) -> tuple[str | None, int | None]:
```

Comment:
these type hints are too loose compared to the error handling logic inside. Please handle all errors in here so you can guarantee that the correct things are returned. This change is what causes all the other changes to be needed, and it makes it harder to track to the source.

Reply:
Fixed it.